compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -168,11 +168,16 @@ class TypeApplications(val self: Type) extends AnyVal {
    *  any type parameter that is-rebound by the refinement.
    */
   final def typeParams(implicit ctx: Context): List[TypeParamInfo] = /*>|>*/ track("typeParams") /*<|<*/ {
+    def isTrivial(prefix: Type, tycon: Symbol) = prefix match {
+      case prefix: ThisType => prefix.cls `eq` tycon.owner
+      case NoPrefix => true
+      case _ => false
+    }
```

Comment:
Can we further refine `isTrivial` to cover the case when `prefix` is static?

Reply:
I am not sure we want to do this. The purpose of `isTrivial` is to fall back to a special case to avoid cyclic reference errors. It seems we don't need to include static `this` types to achieve this. On the other hand, the `isStatic` test could prompt cyclic references itself.